src/java/org/apache/cassandra/io/sstable/format/big/BigTableBlockWriter.java

```diff
@@ -0,0 +1,221 @@
+                bytesRead += count;
+            }
+            out.sync();
+        }
+        catch (IOException e)
+        {
+            throw new FSWriteError(e, out.getPath());
+        }
+    }
+
+    @Override
+    public boolean append(UnfilteredRowIterator partition)
+    {
+        throw new UnsupportedOperationException("Operation not supported by BigTableBlockWriter");
+    }
+
+    @Override
+    public Collection<SSTableReader> finish(long repairedAt, long maxDataAge, boolean openResult)
+    {
+        return finish(openResult);
+    }
+
+    @Override
+    public Collection<SSTableReader> finish(boolean openResult)
+    {
+        setOpenResult(openResult);
+        return finished();
+    }
+
+    @Override
+    public Collection<SSTableReader> finished()
+    {
+        if (finalReader == null)
+            finalReader = SSTableReader.open(descriptor,
+                                             components,
+                                             metadata);
+
+        return ImmutableList.of(finalReader);
+    }
+
+    @Override
+    public SSTableMultiWriter setOpenResult(boolean openResult)
+    {
+        return null;
+    }
+
+    @Override
+    public long getFilePointer()
+    {
+        return 0;
+    }
+
+    @Override
+    public TableId getTableId()
+    {
+        return metadata.id;
+    }
+
+    @Override
+    public Throwable commit(Throwable accumulate)
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.commit(accumulate);
+
+        return accumulate;
+    }
+
+    @Override
+    public Throwable abort(Throwable accumulate)
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.abort(accumulate);
+
+        return accumulate;
+    }
+
+    @Override
+    public void prepareToCommit()
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.prepareToCommit();
+    }
+
+    @Override
+    public void close()
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.close();
+    }
+
+    public void writeComponent(Component.Type type, DataInputPlus in, long size)
+    {
+        logger.info("Writing component {} to {} length {}", type, componentWriters.get(type).getPath(), size);
+        if (in instanceof RebufferingByteBufDataInputPlus)
+            write((RebufferingByteBufDataInputPlus) in, size, componentWriters.get(type));
+        else
+            write(in, size, componentWriters.get(type));
+    }
+
+    private void write(RebufferingByteBufDataInputPlus in, long size, SequentialWriter writer)
+    {
+        logger.info("Block Writing component to {} length {}", writer.getPath(), size);
+
+        try
+        {
+            long bytesWritten = in.consumeUntil(writer, size);
+
+            if (bytesWritten != size)
+                throw new IOException(String.format("Failed to read correct number of bytes from Channel {}", writer));
```

Comment:
{} is not a valid placeholder for `String.format()`.

Reply:
For all of `BTBW` I picked the change from your branch with just a couple minor changes.